### ISSUE TITLE
The UP needs good error messages and warning messages

### ISSUE THREAD

**Issue by jimduff-usds** [CDCgov/prime-reportstream#8496]
### User Story:
As a sender to the U.P., when I send an unsupported message type I get the unhelpful error "Check failed." I should receive a more descriptive message, if possible. 

### Description/Use Case
Our errors and warnings need to tell the customer what the error was (to the best of our understanding) and where the error occurred in their submission.

Submit some FHIR message to the full-elr pipeline that is not OML_O21 or ORU_R01, like ADT-AO1. Confirm the following unhelpful error is returned:

```
 "errors" : [ {
    "scope" : "report",
    "message" : "Check failed.",
    "errorCode" : "UNKNOWN"
  } ],

```
I did various other erroneous submissions, and got the same not-customer-friendly error.

For reference, here's what the Old Pipeline returns if you send ADT-A01 data to it:
```
  "errors" : [ ],
  "warnings" : [ {
    "scope" : "item",
    "indices" : [ 1 ],
    "trackingIds" : [ "message1" ],
    "field" : "ORU_R01",
    "message" : "Unsupported HL7 message type.",
    "errorCode" : "INVALID_HL7_MSG_TYPE_UNSUPPORTED"
  } ],

```
... which is an adequate message. (Not sure why its a warning - that seems like it should be an error)

Bulk Requirement (**this may be out of scope of this ticket, may depend on validation design**): It needs to work with Bulk submissions, clearly showing in which bundle the error occurred.   For bulk FHIR, it needs to reference some sort of customer-created "key", not our internal key.  (For example, in the old pipeline, this requirement is met with the 'trackingId' concept)

### Dev Notes:

Something like the following would be a good start to return:

```
{
    "scope" : "item",
    "indices" : [ 1 ],
    "trackingIds" : [ "message1" ],
    "field" : "ORU_R01",
    "message" : "Unsupported HL7 message type.",
    "errorCode" : "INVALID_HL7_MSG_TYPE_UNSUPPORTED"
  }
```

### Acceptance Criteria
- [ ] The UP returns more detailed error when submitting an unsupported message type.

**Comment by arnejduranovic** [comment]
I expect this story is blocked on the design for how we validate FHIR bundle data. Once the design is in place, we can tackle how we handle, format, and send errors.

ETOR is sending us FHIR, which is fine for them, but we also need to validate and have error messages for HL7 V2. How do we do that? Do we first transform the HL& v2 and validate the FHIR? If we do, those messages probably wont be very helpful to the client sending HL7 V2?

**Comment by bishoyayoub** [comment]
We will need to rewrite this ticket to rectify the one error message (Check Failed). We need to investigate the actual error and apply a more relevant error message. This will also need to be added to the eTor epic.

**Comment by bishoyayoub** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-6182b02547c1130010f459db/issues/CDCgov/prime-reportstream/8496?planning-poker) with Zenhub @arnejduranovic @carlosfelix2 @JessicaWNava @jimmyfagan @mkalish @PatriciaPerozoUSDS @thetaurean @victor-chaparro

**Comment by bishoyayoub** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-6182b02547c1130010f459db/issues/CDCgov/prime-reportstream/8496?planning-poker) with Zenhub @jack-h-wang

**Comment by arnejduranovic** [comment]
Will close this issue as "nothing to fix". There are more specific tickets that were fixed as part of this effort, see PR #8831. As more concrete errors and repro steps are found, we can tackle those separately.
